sphinxgallery/gen_rst.py

```diff
@@ -118,6 +129,31 @@ def _get_data(url):
 get_data = mem.cache(_get_data)
 
 
+def scale_image(in_fname, max_width):
+    """Scale image such that width <= max_width
+    """
+    try:
+        from PIL import Image
+    except ImportError:
+        import Image
+    img = Image.open(in_fname)
+    width_in, height_in = img.size
+
+    if width_in <= max_width:
```

Comment:
this is possibly the only extra clause that distinguishes this function, from the function that does the thumbnails.
Is there a reason to use this instead than the `make_thumbnail` with different input parameters and add there a clause like 
```
if scale >1.0:
    return
```